### PR TITLE
Vl cv integration

### PR DESCRIPTION
**Pull request by 1999michael**

### CODE REVIEW

File: cv/cv/launch/pipeline.launch [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <launch>
     <!-- Launch zed camera -->
-    <include file="$(find zed_wrapper)/launch/zed.launch"/>
+    <!--<include file="$(find zed_wrapper)/launch/zed.launch"/>-->
```
Comment (kajananchinniah) [inline cv/cv/launch/pipeline.launch:3]: Restore this back to master?

File: cv/cv/launch/pipeline.launch [hunk 2]
```diff
@@ -9,13 +9,13 @@
         type="zed_wrapper_server.py"
         output="screen"
     >
-        <remap from="image" to="/zed/zed_node/rgb/image_rect_color"/>
+        <remap from="image" to="/zed/zed_node/left/image_rect_color"/>
```
Comment (kajananchinniah) [inline cv/cv/launch/pipeline.launch:12]: Keep what was on master

File: cv/cv/src/cv_utils/camera_projection.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def __init__(self):
             self.depth_values += [self.depth_map[key]]
         self.depth_values = np.array(self.depth_values)
 
-        camera_info = rospy.wait_for_message('/zed/zed_node/rgb/camera_info', CameraInfo)
+        camera_info = rospy.wait_for_message('/zed/zed_node/left/camera_info', CameraInfo)
```
Comment (kajananchinniah) [inline cv/cv/src/cv_utils/camera_projection.py:29]: Keep what's on master

File: cv/lane_detection/CMakeLists.txt [hunk 4]
```diff
@@ -6,11 +6,23 @@ find_package(catkin REQUIRED COMPONENTS
...
+  DEPENDENCIES
+  geometry_msgs
+)
+
 # Generate cmake config files via catkin_package
 catkin_package(
   INCLUDE_DIRS
-  CATKIN_DEPENDS
+  CATKIN_DEPENDS message_runtime
   )
```
Comment (kajananchinniah) [inline cv/lane_detection/CMakeLists.txt:27]: Fix end of fild

File: cv/lane_detection/src/lane_detection_model_inference.py [hunk 5]
```diff
@@ -117,7 +117,8 @@ def find_edge_channel(img):
 
 
 def get_input(frame):
-    frame = cv2.resize(frame,(1280,720),interpolation=cv2.INTER_AREA)
+    #print("HELLO")
```
Comment (kajananchinniah) [inline cv/lane_detection/src/lane_detection_model_inference.py:120]: @spencerteetaert for vis

Why is this necessary?

File: cv/lane_detection/src/lane_detection_wrapper_client.py [hunk 6]
```diff
@@ -11,7 +11,7 @@
 from sensor_msgs.msg import Image
 
 from cv.msg import FloatArray, FloatList
-from geometry_msgs.msg import Point
+from geometry_msgs.msg import PoseArray, Point
```
Comment (kajananchinniah) [inline cv/lane_detection/src/lane_detection_wrapper_client.py:14]: Do you need the pose array message?

File: description/urdf/constants.xacro [hunk 7]
```diff
@@ -108,11 +108,11 @@
     <xacro:property name="ZED_camera_length" value="0.175"/>
     <xacro:property name="ZED_camera_width" value="0.03"/>
     <xacro:property name="ZED_camera_height" value="0.033"/>
-    <xacro:property name="ZED_camera_pitch" value="${15 * pi/180}"/> <!-- 19.66 degrees ensures the camera's field of view overlaps with the end of the lidar readings-->
+    <xacro:property name="ZED_camera_pitch" value="${50 * pi/180}"/> <!-- 19.66 degrees ensures the camera's field of view overlaps with the end of the lidar readings-->
```
Comment (kajananchinniah) [inline description/urdf/constants.xacro:111]: Delete the stale comment at the side

File: description/urdf/constants.xacro [hunk 8]
```diff
@@ -142,7 +142,7 @@
     <xacro:property name="ref_lon" value="-79.3903"/>
     <xacro:property name="ref_alt" value="76.0000"/>
 
-    <xacro:property name="gps_drift" value="1.5 1.5 1.5"/>
+    <xacro:property name="gps_drift" value="0 0 0"/>
```
Comment (kajananchinniah) [inline description/urdf/constants.xacro:145]: GPS drift shouldn't be zero, restore them to default value

File: description/urdf/constants.xacro [hunk 9]
```diff
@@ -142,7 +142,7 @@
     <xacro:property name="ref_lon" value="-79.3903"/>
     <xacro:property name="ref_alt" value="76.0000"/>
 
-    <xacro:property name="gps_drift" value="1.5 1.5 1.5"/>
```
Comment (kajananchinniah) [inline description/urdf/constants.xacro:145]: Action item on @spencerteetaert to check the zeds image resolution

File: nav/cv_pkg/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,205 @@
+cmake_minimum_required(VERSION 3.0.2)
```
Comment (kajananchinniah) [inline nav/cv_pkg/CMakeLists.txt:1]: Could you delete the test pkg since we aren't going to be using it ?

File: nav/filter_lidar_data/src/ignore_lidar_node.cpp [hunk 11]
```diff
@@ -32,10 +32,10 @@ IgnoreLidarNode::IgnoreLidarNode(ros::NodeHandle nh)
     
     // Set rectangular bounds where lidar data needs to be adjusted. Based on the 2nd and 3rd waypoint
     // TODO: Tune padding  
-    bounds_[0].x = second_pose_stamped.pose.position.x + 1.5;
-    bounds_[0].y = second_pose_stamped.pose.position.y + 4.0;
-    bounds_[1].x = third_pose_stamped.pose.position.x - 1.0;
-    bounds_[1].y = third_pose_stamped.pose.position.y - 4.0;
+    bounds_[0].x = second_pose_stamped.pose.position.x + 3.0;
```
Comment (kajananchinniah) [inline nav/filter_lidar_data/src/ignore_lidar_node.cpp:35]: Is this change intentional?

File: nav/virtual_layers/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+find_package(catkin REQUIRED COMPONENTS
+  costmap_2d
+  dynamic_reconfigure
+  roscpp
+  rospy
+  std_msgs
+  tf
+  geometry_msgs #for topic subscription
+  cv_pkg
```
Comment (kajananchinniah) [inline nav/virtual_layers/CMakeLists.txt:12]: We have a dependency on the CV pkg (the one @spencerteetaert made) not cv_pkg (the test pkg) I believe

File: nav/virtual_layers/package.xml [hunk 13]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>virtual_layers</name>
+  <version>0.0.0</version>
+  <description>The virtual layers package</description>
+
+  <maintainer email="kajanan@todo.todo">kajanan</maintainer>
```
Comment (kajananchinniah) [inline nav/virtual_layers/package.xml:7]: Change my name to your name

File: odom/config/odom_local.yaml [hunk 14]
```diff
@@ -28,15 +28,6 @@ odom0_queue_size: 10
 odom0_differential: false
 odom0_relative: false
 
-odom1: zed/zed_node/odom
```
Comment (kajananchinniah) [inline odom/config/odom_local.yaml:31]: Is the 2d mode and removal of zed as an Odom source intentional?

Comment (1999michael) [reply odom/config/odom_local.yaml:31]: We'll need it regardless if we aren't allowed to tilt the camera back from 50 deg. I have no idea if the camera is adjustable nor are we going to

File: nav/filter_lidar_data/src/ignore_lidar_node.cpp (hunk 11 above)
Comment (1999michael) [reply nav/filter_lidar_data/src/ignore_lidar_node.cpp:35]: to increase the bounds (max GPS offset we estimate is like, 2m, so we did 3m in all 4 cardinal directions). I think this change shouldn't leave an impact since ivy is changing it anyways, and I changed local/global costmaps to subscribe from /scan rather than /scan_modified

**Review by kajananchinniah — APPROVED** [review]
We need this soon so approved
